.github/workflows/ci.yaml

```diff
@@ -66,13 +115,9 @@ jobs:
         restore-keys: |
           cabal-${{ runner.os }}-${{ hashFiles('cabal.project', 'default.nix', 'shell.nix') }}
 
-    - name: 🧰 Prepare tools
-      run: |
-        nix develop .#ci --command bash -c 'cabal update'
-
     - name: 🔨 Build
       run: |
-        nix develop .#ci --command bash -c 'cabal build ${{ matrix.package }}'
+        nix develop .#ci --command bash -c 'cabal update && cabal build ${{ matrix.package }}'
```

Comment:
I think we can drop this step as build errors would already appear in the previous job and `cabal test` would also build anything that was not yet built (e.g. benchmarks are not built when we do `cabal build all`)

Reply:
I'd like to keep it that way to limit inter jobs dependencies.

This P.R. proposes an improvement by just preparing the caches before running the existing job. So if we decide to drop the warm up the caches job this job still works and make sense.

/cc @ch1bo